syntax = "proto3";

package guru.qa.grpc.niffler;
import "google/protobuf/empty.proto";

service NifflerCurrencyService {
    rpc GetAllCurrencies(google.protobuf.Empty) returns (CurrencyResponse);
    rpc CalculateRate(CalculateRequest) returns (CalculateResponse);
}
message CurrencyResponse {
    repeated Currency allCurrencies = 1;
}
message Currency {
    optional CurrencyValues currency = 1;
    optional double currencyRate = 2;
}
message CalculateRequest {
    optional CurrencyValues spendCurrency = 1;
    optional CurrencyValues desiredCurrency = 2;
    optional double amount = 3;
}
message CalculateResponse {
    optional double calculatedAmount = 1;
}
enum CurrencyValues {
    UNSPECIFIED = 0;
    RUB = 1;
    USD = 2;
    EUR = 3;
    KZT = 4;
}